scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/task/Task.java

```diff
@@ -725,39 +727,44 @@ public String toString() {
     public String display() {
         String nl = System.lineSeparator();
 
-        return "Task \'" + name + "\' : " + nl + "\tDescription = '" + description + "'" + nl +
-               (restartTaskOnError.isSet() ? "\trestartTaskOnError = '" + restartTaskOnError.getValue() + '\'' + nl
-                                           : "") +
-               (onTaskError.isSet() ? "\tonTaskError = '" + onTaskError.getValue() + '\'' + nl : "") +
-               (maxNumberOfExecution.isSet() ? "\tmaxNumberOfExecution = '" +
-                                               maxNumberOfExecution.getValue().getIntegerValue() + '\'' + nl
-                                             : "") +
-               "\ttag = " + tag + nl + "\tvariables = {" + nl +
-               Joiner.on('\n').withKeyValueSeparator("=").join(variables) + nl + "}" + nl + "\tgenericInformation = {" +
-               nl + Joiner.on('\n').withKeyValueSeparator("=").join(genericInformation) + nl + "}" + nl +
-               "\tInputFiles = " + inputFiles + nl + "\tOutputFiles = " + outputFiles + nl +
-               "\tParallelEnvironment = " + parallelEnvironment + nl + "\tFlowBlock = '" + flowBlock + "'" + nl +
-               "\tSelectionScripts = " + displaySelectionScripts() + nl + "\tForkEnvironment = " + forkEnvironment +
-               nl + "\tPreScript = " + ((preScript != null) ? preScript.display() : null) + nl + "\tPostScript = " +
-               ((postScript != null) ? postScript.display() : null) + nl + "\tCleanScript = " +
-               ((cScript != null) ? cScript.display() : null) + nl + "\tFlowScript = " +
-               ((flowScript != null) ? flowScript.display() : null) + nl + "\tPreciousResult = " + preciousResult + nl +
-               "\tPreciousLogs = " + preciousLogs + nl + "\tRunAsMe = " + runAsMe + nl + "\tWallTime = " + wallTime +
-               nl + "\tDependences = " + dependences;
+        return "Task \'" + name + "\' : " +
```

Comment:
This concatenation is still a bit difficult to follow. What about creating a method for each component? Like `String appendMaxNumberOfExecution(...)` `String appendInputFiles(...)`. Not sure it will be easier to read.

Reply:
Sure, this concatenation is crazy. I will see what I can do